src/filesystem/index.ts

```diff
@@ -106,6 +106,20 @@ const WriteFileArgsSchema = z.object({
   content: z.string(),
 });
 
+const EditOperation = z.object({
+  // The text to search for
+  oldText: z.string().describe('Text to search for - can be a substring of the target'),
+  // The new text to replace with
+  newText: z.string().describe('Text to replace the found text with'),
+  // Optional: preview changes without applying them
+  dryRun: z.boolean().default(false).describe('Preview changes using git-style diff format')
+});
+
+const EditFileArgsSchema = z.object({
+  path: z.string(),
+  edits: z.array(EditOperation),
```

Comment:
IMO `dryRun` makes more sense as an argument here, rather than per-operation.

Reply:
I'll take a look at this while working on the other dry run suggestion, thanks.